appinventor/appengine/src/com/google/appinventor/server/project/youngandroid/YoungAndroidProjectService.java

```diff
@@ -911,4 +1046,94 @@ public RpcResult getBuildResult(User user, long projectId, String target) {
   public int getCurrentProgress(User user, long projectId, String target) {
     return storageIo.getBuildStatus(user.getUserId(), projectId);
   }
+
+  /**
+   * This method reads from a stream based on the passed connection. It reads
+   * the content as bytes, so it can deal with binary files
+   *
+   * @param connection the connection to read from
+   * @return the contents of the stream
+   * @throws IOException if it cannot read from the http connection
+   */
+  private static byte[] getResponseBytes(HttpURLConnection connection) throws IOException {
+    // Use the content encoding to convert bytes to characters.
+    InputStream input = connection.getInputStream();
+    int bytesRead = 0;
+    int contentLength = connection.getContentLength();
```

Comment:
We should make a note for the future that this function should only be called for servers we trust as using the Content-Length header could be problematic with untrustworthy servers.

Reply:
Understood, but not changed for now.